bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/Main.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.beam;
+
+import com.google.bigtable.repackaged.com.google.api.core.InternalApi;
+import com.google.bigtable.repackaged.com.google.api.core.InternalExtensionOnly;
+import com.google.cloud.bigtable.beam.hbasesnapshots.ImportJobFromHbaseSnapshot;
+import com.google.cloud.bigtable.beam.sequencefiles.CreateTableHelper;
+import com.google.cloud.bigtable.beam.sequencefiles.ExportJob;
+import com.google.cloud.bigtable.beam.sequencefiles.ImportJob;
+import java.io.File;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+
+/** Entry point for create-table/import/export job submission. */
+@InternalExtensionOnly
+public class Main {
+  /** For internal use only - public for technical reasons. */
+  @InternalApi("For internal usage only")
+  public Main() {}
+
+  public static void main(String[] args) throws Exception {
+    if (args.length < 1) {
+      usage();
+      System.exit(1);
+    }
+
+    String[] subArgs = Arrays.copyOfRange(args, 1, args.length);
+
+    switch (args[0]) {
+      case "export":
+        ExportJob.main(subArgs);
+        break;
+      case "import":
+        ImportJob.main(subArgs);
+        break;
+      case "importsnapshot":
+        ImportJobFromHbaseSnapshot.main(subArgs);
+        break;
+      case "create-table":
+        CreateTableHelper.main(subArgs);
+        break;
+      default:
+        usage();
+        System.exit(1);
+    }
+  }
+
+  private static void usage() {
+    String jarName;
```

Comment:
Since the initialization could through an exception, hence in the code this is actually initialized twice, making it final will not compile, are you suggesting using a temp variable in the try block? IMO that would make the logic more complex.
Did I miss anything?

Reply:
I dont quite understand what you mean here. The variable will only be initialized once: either in the try or the catch. Marking it final makes sure that in the future someone doesnt accidentally remove the assignment in the catch

Reply:
Since we initialize `jarName`  within a try/catch statement, the compiler think this variable could get assigned twice and will throw a `variable jarName might already have been assigned` error.

I did some google search, the best workaround is to use a second variable before assignment or use a function like:

```
    final String jarName = getJarName();

    String getJarName() {
        String temp;
        try {
            temp = ...
        } catch (Exception e) {
            temp = ...
        }
        return temp
    }
```

Which I find hardly any better...

Please advise if I have misunderstood this.